.github/workflows/test_asterinas.yml

```diff
@@ -45,17 +90,32 @@ jobs:
       # TODO: add component check.
 
   integration-test:
-    if: github.event_name == 'push' || github.event_name == 'pull_request'
+    needs: build-initrd
     runs-on: ubuntu-latest
     timeout-minutes: 30
     container:
       image: asterinas/asterinas:0.8.0
       options: --device=/dev/kvm
+    env:
+      PREBUILT_INITRAMFS: result/initrd.gz
     steps:
       - run: echo "Running in asterinas/asterinas:0.8.0"
 
       - uses: actions/checkout@v4
 
+      - name: Download Initrd
+        uses: actions/download-artifact@v4
+        with:
+          name: initrd.gz
+          path: result
+
+        # OSDK keep checking the options that have been overridden
+        # https://github.com/asterinas/asterinas/issues/1237
+      - name: Workaround for OSDK
+        run: |
+          mkdir -p test/build
+          touch test/build/initramfs.cpio.gz
+
```

Comment:
We support building initramfs both by Nix and by original Makefile, right?

I think you can cover both cases in the CI. For example, one test uses Nix-built initramfs, and others uses initramfs built by Makefile